backend/service/aws/aws.go

```diff
@@ -62,47 +77,90 @@ func New(cfg *anypb.Any, logger *zap.Logger, scope tally.Scope) (service.Service
 	}
 
 	ds := getScalingLimits(ac)
-
 	awsHTTPClient := &http.Client{}
+	awsClientCommonOptions := []func(*config.LoadOptions) error{
```

Comment:
nit: use `LoadOptionsFunc`

Reply:
Ill try again but the compiler was not having it, thats what I originally lead with.

Reply:
```diff
➜ clutch git:(aws-multi-account-gen-clients) ✗ git --no-pager diff
diff --git a/backend/service/aws/aws.go b/backend/service/aws/aws.go
index 36fc93e9..18f90c5c 100644
--- a/backend/service/aws/aws.go
+++ b/backend/service/aws/aws.go
@@ -79,7 +79,7 @@ func New(cfg *anypb.Any, logger *zap.Logger, scope tally.Scope) (service.Service

 	ds := getScalingLimits(ac)
 	awsHTTPClient := &http.Client{}
-	awsClientCommonOptions := []func(*config.LoadOptions) error{
+	awsClientCommonOptions := []config.LoadOptionsFunc{
 		config.WithHTTPClient(awsHTTPClient),
 		config.WithRetryer(func() aws.Retryer {
 			customRetryer := retry.NewStandard(func(so *retry.StandardOptions) {
@@ -107,7 +107,7 @@ func New(cfg *anypb.Any, logger *zap.Logger, scope tally.Scope) (service.Service
 	return c, nil
 }

-func (c *client) configureAdditionalAccountClient(accounts []*awsv1.AWSAccount, ds *awsv1.ScalingLimits, awsHTTPClient *http.Client, awsClientOptions []func(*config.LoadOptions) error) error {
+func (c *client) configureAdditionalAccountClient(accounts []*awsv1.AWSAccount, ds *awsv1.ScalingLimits, awsHTTPClient *http.Client, awsClientOptions []config.LoadOptionsFunc) error {
 	for _, account := range accounts {
 		accountRoleARN := fmt.Sprintf("arn:aws:iam::%s:role/%s", account.AccountNumber, account.IamRole)
 		// For doing STS calls it does not matter which region client we are using, as they are not bounded by region
```

```
building...
# github.com/lyft/clutch/backend/service/aws
service/aws/aws.go:94:10: cannot use append(awsClientCommonOptions, "github.com/aws/aws-sdk-go-v2/config".WithRegion(region)) (type []"github.com/aws/aws-sdk-go-v2/config".LoadOptionsFunc) as type []func(*"github.com/aws/aws-sdk-go-v2/config".LoadOptions) error in argument to "github.com/aws/aws-sdk-go-v2/config".LoadDefaultConfig
service/aws/aws.go:121:11: cannot use append(awsClientOptions, "github.com/aws/aws-sdk-go-v2/config".WithRegion(region)) (type []"github.com/aws/aws-sdk-go-v2/config".LoadOptionsFunc) as type []func(*"github.com/aws/aws-sdk-go-v2/config".LoadOptions) error in argument to "github.com/aws/aws-sdk-go-v2/config".LoadDefaultConfig
failed to build, error: exit status 2
```